backend/app/core/service/loader.py

```diff
@@ -0,0 +1,150 @@
+from fastapi import (
+    HTTPException,
+)
+from sqlalchemy.exc import IntegrityError
+from sqlalchemy.orm import Session
+
+from app.db.crud.document import create_document
+from app.db.schemas.metadata import DocumentCreateWithMetadata
+from app.db.models import (
+    Document,
+    Event,
+    Sector,
+    Hazard,
+    Response,
+    DocumentSector,
+    DocumentHazard,
+    DocumentResponse,
+    Framework,
+    DocumentFramework,
+    Instrument,
+    DocumentInstrument,
+    DocumentLanguage,
+)
+from sqlalchemy.dialects.postgresql import insert
+
+
+def persist_document_and_metadata(
+    db: Session,
+    document_with_metadata: DocumentCreateWithMetadata,
+    creator_id: int,
+):
+    try:
+        db_document = create_document(db, document_with_metadata.document, creator_id)
+        write_metadata(db, db_document, document_with_metadata)
+        db.commit()
+
+        return db_document
+    except Exception as e:
+        if isinstance(e, IntegrityError):
+            raise HTTPException(409, detail="Document already exists")
+        raise e
+
+
+def write_metadata(
+    db: Session,
+    db_document: Document,
+    document_with_metadata: DocumentCreateWithMetadata,
+):
+    # doc language
+    for language_id in document_with_metadata.language_ids:
+        db_doc_lang = DocumentLanguage(
+            language_id=language_id,
+            document_id=db_document.id,
+        )
+        db.add(db_doc_lang)
+
+    # events
+    for event in document_with_metadata.events:
+        db_event = Event(
+            document_id=db_document.id,
+            name=event.name,
+            description=event.description,
+        )
+        db.add(db_event)
+
+    # sectors
+    for meta in document_with_metadata.sectors:
+        insert_stmt = insert(Sector).values(
```

Comment:
These inserts are a little surprising, I thought we were following a pattern of making sure all valid values were already present in the tables and we would use the uniqueness constraint to look them up...?

Reply:
Yes, ultimately. 
At the moment, the names on those tables are unique, so we end up with a small subset anyway.
Once we have the migrations for the lookups, perhaps we can look at using lookups then?